website/project/model.py

```diff
@@ -3219,11 +3221,50 @@ def add_unregistered_contributor(self, fullname, email, auth,
 
         self.add_contributor(
             contributor, permissions=permissions, auth=auth,
-            log=True, save=False,
+            send_email=send_email, log=True, save=False
         )
         self.save()
         return contributor
 
+    def add_contributor_registered_or_not(self, auth, user_id=None, full_name=None, email=None,
+                                          send_email='default', permissions=None, bibliographic=True, save=False):
```

Comment:
If this is only being called by the API, probably best to `send_email='false'` by default.